nibabel/nicom/utils.py

```diff
@@ -31,15 +29,20 @@ def find_private_section(dcm_data, group_no, creator):
     """
     if hasattr(creator, 'search'):
         match_func = creator.search
-    else:  # assume string / bytes
-        match_func = asstr(creator).__eq__
+    else:
+        if isinstance(creator, bytes):
+            creator = creator.decode('latin-1')
```

Comment:
This one has both `bytes` and `str` tests in `nicom/tests/test_utils.py` so I assume it's necessary